core/src/main/java/org/apache/accumulo/core/file/FileOperations.java

```diff
@@ -267,6 +271,10 @@ public Set<ByteSequence> getColumnFamilies() {
     public boolean isRangeInclusive() {
       return inclusive;
     }
+
+    public Range getFence() {
```

Comment:
Instead of adding this getFence method would is be possible to pass TabletFile to this class instead of a String?  Then the getFence() and getFilename() methods could be replaced with a getTabletFile() method that has the filename and range.

Reply:
So looking at this I'm not sure the best way to handle the fence/range here in `FileOperations`. FileOperations is actually abstract and [MapFileOperations](https://github.com/apache/accumulo/blob/18e7a391d0eee40ee7258cd6bdc859a8cfde649a/core/src/main/java/org/apache/accumulo/core/file/map/MapFileOperations.java) currently extends it (although might be removed by ##3359). So arguably the TabletFile or Fence reference really needs to go into the subclass which is `RFileOperations`.

Reply:
I was thinking that FileOperations could look at the TabletFile and open and RFile or MapFile (not passing TabletFile down any further) and optionally (if the range is present or not empty.) wrap it with a fencing iterator.  So file operations is where the file+optional range is handled.

Reply:
So the next problem is all the validation done by TabletFile. I was trying to just replace filename with TabletFile everywhere and require those objects but that breaks quite a bit of things. The reason is TabletFile does a lot of validation in the constructor to ensure the correct directory structure and lots of places (mostly tests) are not passing in rfiles that pass validation because they are temp files.  For example https://github.com/apache/accumulo/blob/ba472d6e24daa8f0014a22cabace3061f5d46413/core/src/test/java/org/apache/accumulo/core/client/rfile/RFileClientTest.java#L230

I also don't know if any of the Map file stuff would pass validation either (I doubt it) but there are no tests for that to even try it unless I write some.

So not sure the past approach. I think the end goal would be to only support TabletFile to prevent errors so we'd need to create another implementation of TabletFile that doesn't do validation checks (maybe just for testing). But for the short term for now maybe we just could have FileOperations support passing either a TabletFile or a filename and then we can track both and make sure only 1 type is set. 

But I'm good with doing whatever, it really just depends how much code we want to preemptively change right now for places that are actually using FileOperations vs waiting until we are storing the new ranges in metadata and can actually use the data and make the change then.